inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java

```diff
@@ -81,7 +82,7 @@
     private static final String ORACLE_CLOUD_ASSET_TAG_FILE = "/sys/devices/virtual/dmi/id/chassis_asset_tag";
     private static final String ORACLE_CLOUD_WINDOWS_ASSET_TAG_CMD = "wmic systemenclosure get smbiosassettag";
     private static final String DO_SYS_VENDOR_FILE = "/sys/devices/virtual/dmi/id/sys_vendor";
-    private static final Boolean DEDUCE_ENVIRONMENT_DEFAULT = true;
+    private static final Boolean DEDUCE_ENVIRONMENT_DEFAULT = GraphicsEnvironment.isHeadless();
```

Comment:
This is up for discussion

Reply:
I don't think we should depend on AWT

We can make a decision to default to `false` for 3.0 or at least default to not make the network calls for certain clouds

Reply:
Perhaps move this change and any other changes related to environment deduction to a separate PR so we can merge the rest of this?

Reply:
Agreed

Reply:
I have removed it for now